### PR TITLE
Fix #3651 - remove pressure value on Highlighter strokes first point

### LINKED ISSUE #3651: The first point of Highlighter strokes has a pressure value
**Affects versions :**
 - OS: all
 - Version of Xournal++: `release-1.1` since #3344 and `master` since #3644.

**Describe the bug**
On a pressure sensitive device (or with pressure inference), the first point of highlighter strokes gets a pressure value assigned. This leads to `stroke::hasPressure` to return true, thus messing up saved files by adding a useless array of pressure values.

**To Reproduce**
Create a highlighter stroke with a pressure sensitive device, save and inspect the file.

A fix is coming soon :-)

### PR DESCRIPTION
**Pull request by bhennion**
Merging in `release-1.1` in 24h unless someone objects.

This will also need to be added to master ASAP: the bug corrupts files (with little harm, but still). How should I do that? Another merge, cherry-picking?

### CODE REVIEW

**Comment by Febbe** [comment]
Rebase this onto release-1.1, then merge release-1.1 into master

File: src/control/tools/StrokeHandler.cpp [hunk 1]
```diff
@@ -369,9 +369,14 @@ void StrokeHandler::onButtonPressEvent(const PositionInputData& pos) {
...
-        createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
+        this->hasPressure = this->xournal->getControl()->getToolHandler()->getToolType() == TOOL_PEN &&
+                            pos.pressure != Point::NO_PRESSURE;
 
-        this->hasPressure = this->stroke->getToolType() == STROKE_TOOL_PEN && pos.pressure != Point::NO_PRESSURE;
+        if (this->hasPressure) {
+            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
+        } else {
+            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y));
+        }
```
Comment (Febbe) [inline src/control/tools/StrokeHandler.cpp:379]: What's the difference to this?:
```suggestion
            createStroke(Point(this->buttonDownPoint.x, this->buttonDownPoint.y, pos.pressure));
```

Comment (bhennion) [reply src/control/tools/StrokeHandler.cpp:379]: When the tool is the highlighter, then the event `pos` may have a pressure value, but we do not want to give the point that pressure value.
The boolean `this->hasPressure` is always `false` when the tool is the highlighter, and `true` when the tool is the normal pen and the device is pressure sensitve.

Comment (Febbe) [reply src/control/tools/StrokeHandler.cpp:379]: Ok, looked into the source of the function.
Wouldn't it an idea to move that into it? Then the pressure is not set wrongly at other places?

Comment (Febbe) [reply src/control/tools/StrokeHandler.cpp:379]: If not this LGTM and you can merge it ASAP.

**Comment by Febbe** [comment]
LGTM

**Pull request merged**